javagen/src/main/java/com/azure/autorest/template/ConvenienceMethodTemplateBase.java

```diff
@@ -359,7 +359,11 @@ protected void addImports(Set<String> imports, List<ConvenienceMethod> convenien
     }
 
     protected void addGeneratedAnnotation(JavaType typeBlock) {
-        typeBlock.annotation(Annotation.GENERATED.getName());
+        if (JavaSettings.getInstance().isBranded()) {
+            typeBlock.annotation(Annotation.GENERATED.getName());
+        } else {
+            typeBlock.annotation(Annotation.METADATA.getName() + "(generated = true)");
```

Comment:
I may be lack of context on this. What is `@MetaData` used for besides setting `generated`?

Reply:
It can be used for any metadata we want to add about a method, class or property. Right now, we are using it to mark methods as generated and classes as either fluent or immutable. In the future, there might be other things added to this annotation.

Reply:
On the chat.

Maybe we could use a full class name here "com.generic.core.annotation.Metadata(generated = true)", and not add the "import com.generic.core.annotation.Metadata"?

The "Metadata" may conflict with existing code https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/resourcemanager/azure-resourcemanager-monitor/src/main/java/com/azure/resourcemanager/monitor/models/DataCollectionRuleMetadata.java#L13

Reply:
Or if it is urgent, you may add Metadata into the reserved word list.
https://github.com/Azure/autorest.java/blob/main/javagen/src/main/java/com/azure/autorest/util/CodeNamer.java#L65-L81
https://github.com/Azure/autorest.java/blob/main/preprocessor/src/main/java/com/azure/autorest/preprocessor/namer/CodeNamer.java#L63-L79

Reply:
Here https://github.com/Azure/autorest.java/pull/2608/commits/a6c34f6dbe4c4442ffd599cc3e614ad34aae3ceb

Reply:
Updated code to add the metadata annotation import only for unbranded libraries.